pyaedt/application/Design.py

```diff
@@ -2035,18 +2035,16 @@ def delete_design(self, name=None, fallback_design=None):
             name = self.design_name
         self.oproject.DeleteDesign(name)
         try:
-            if fallback_design:
-                new_designname = fallback_design
-            else:
-                if not self._oproject.GetActiveDesign():
-                    new_designname = None
-                    self.odesign = None
-                else:
-                    new_designname = self._oproject.GetActiveDesign().GetName()
-            self.set_active_design(new_designname)
+            if not self._oproject.GetActiveDesign():
+                self.odesign = None
         except:
             pass
-
+        if fallback_design and (fallback_design in
+                                [x for i in list(self._oproject.GetDesigns()) for x in (i.GetName(), i.GetName()[2:])]):
+            try:
+                self.set_active_design(fallback_design)
+            except:
+                pass
```

Comment:
Should we not be returning ``False`` here?

Reply:
I was thinking the same.
I have refactored the method.